src/operator.test.js

```diff
@@ -27,44 +28,32 @@ describe("operator", () => {
             "cert-manager.io/certificate-name": "tls"
           }
         }
-      }
-    };
-
-    await this.watchResourceMock.callsArgWith(3, invalidTestEvent);
-
-    await this.operator.start();
-
-    expect(this.secretHandlerBackupMock).to.not.have.been.called;
-  });
-
-  it("should not call secretHandlerBackupMock if the event type is ResourceEventType.Added", async () => {
-    const invalidTestEvent = {
-      object: {
-        metadata: {}
       },
-      type: ResourceEventType.Added
+      type: ResourceEventType.Modified
     };
 
-    await this.watchResourceMock.callsArgWith(3, invalidTestEvent);
+    this.watchResourceMock.callsArgWith(3, invalidTestEvent);
 
     await this.operator.start();
 
     expect(this.secretHandlerBackupMock).to.not.have.been.called;
   });
 
-  it("should not call secretHandlerBackupMock if the event type is ResourceEventType.Deleted", async () => {
-    const invalidTestEvent = {
-      object: {
-        metadata: {}
-      },
-      type: ResourceEventType.Deleted
-    };
+  [ResourceEventType.Added, ResourceEventType.Deleted].forEach((type) => {
+    it(`should not create a backup when the event type is ResourceEventType.${type}`, async () => {
+      const invalidTestEvent = {
+        object: {
+          metadata: {}
+        },
+        type
+      };
```

Comment:
I was thinking that we could include all tests that require no backup in this loop. This would require the full events to be listed in the array.
I think that would be cleaner, since adding one of these kinds of tests would only require lising the even that should not trigger a backup in the list.